lib/index.js

```diff
@@ -21,22 +23,52 @@ exports.plugin = {
         const p2p = container.resolvePlugin('p2p')
 
         container.resolvePlugin('p2p').getMonitor().broadcastTransactions = async transactions => {
-            const originalLength = transactions.length
-
             // Filter transactions, based on the public keys in the config
-            transactions = transactions.filter(transaction => !options.publicKeys.includes(transaction.data.senderPublicKey))
+            transactions = transactions.reduce((transactions, transaction) => {
+                if (options.publicKeys.includes(transaction.data.senderPublicKey)) {
+                    transactions.filtered.push(transaction)
+                } else {
+                    transaction.unfiltered.push(transaction)
+                }
+                return transactions
+            }, {
+                filtered: [],
+                unfiltered: [],
+            })
 
-            const lengthDiff = originalLength - transactions.length
-            if (lengthDiff) {
-                logger.debug(`[VANIR] Filtered ${pluralize(
-                    'transaction',
-                    lengthDiff,
-                    true,
-                )} to self-forge`)
-            }
+            if (transactions.filtered.length) {
+                let delegates = options.delegates || []
+
+                if (!Array.isArray(delegates)) {
+                    delegates = [delegates]
+                }
 
-            if (!transactions.length) {
-                return;
+                let hasActiveDelegates = !delegates.length
```

Comment:
if the delegate array is empty, this will evaluate to `true` and you end up with `hasActiveDelegates` being true which results in the transactions being filtered without doing the proper delegate check. Is this intended behaviour?

Reply:
Yes that is intended. No configured delegates -> always filter.

Reply:
In other words, when no delegates are configured it fallsback to the current behaviour.